argus/scenarios/service_mock.py

```diff
@@ -268,3 +271,33 @@ def x509():
             TO9FwPXIRcR/XrcMDn7slaHtbILM3P4pIbaXUhvel+qLOAMp6k2iDl2Q
             -----END CERTIFICATE-----
             """.strip())
+
+
+class HTTPMetadataServiceApp(BaseServiceApp):
+    """Custom OpenStack http metadata."""
+
+    METADATA = {
+        "keys": [
+            {
+                "name": "argus_cert",
+                "type": "x509",
+                "data": util.get_certificate()
+            }
+        ] + [{
+            "name": "argus_key",
+            "type": "ssh",
+            "data": data
+        } for data in util.get_public_keys()],
+        "admin_pass": "PASsw0r4&!="
+    }
+
+    @cherrypy.expose
+    def default(self, *args):
+        link = "/".join(args)
+        if "latest/meta_data.json" not in link:
+            # Handle invalid and password posting cases.
+            raise cherrypy.HTTPError(404)
+        # Fill in the metadata password provided by the config file.
+        key = "admin_pass"
+        self.METADATA[key] = self.scenario.get_metadata()[key]
```

Comment:
Don't modify it like this. Instead, implement a method which returns a dictionary and in that method call scenario.get_metadata().